packages/xo-web/src/xo-app/logs/index.js

```diff
@@ -119,31 +121,40 @@ const UNHEALTHY_VDI_CHAIN_ERROR = 'unhealthy VDI chain'
 const UNHEALTHY_VDI_CHAIN_LINK =
   'https://xen-orchestra.com/docs/backup_troubleshooting.html#vdi-chain-protection'
 
+const filterOptionRenderer = ({ label, value }) =>
+  _(label, message => <option value={value}>{message}</option>)
+
 class Log extends BaseComponent {
   state = {
-    filter: 'all',
+    filter: DEFAULT_CALL_FILTER,
   }
 
-  render () {
-    const { props, state } = this
-    const predicate = PREDICATES[state.filter]
+  _getFilteredCalls = () =>
+    filter(this.props.log.calls, PREDICATES[this.state.filter.value])
+
+  _filterValueRenderer = ({ label, value }) =>
+    _(label, message => (
+      <option value={value}>
+        {message} ({this._getFilteredCalls().length})
+      </option>
+    ))
 
+  render () {
     return (
       <div>
-        <select
-          className='form-control'
+        <Select
+          labelKey='label'
           onChange={this.linkState('filter')}
-          value={state.filter}
-        >
-          {map(CALL_FILTER_OPTIONS, ({ label, value }) =>
-            _({ key: value }, label, message => (
-              <option value={value}>{message}</option>
-            ))
-          )}
-        </select>
+          optionRenderer={filterOptionRenderer}
+          options={CALL_FILTER_OPTIONS}
+          required
+          value={this.state.filter}
+          valueKey='value'
+          valueRenderer={this._filterValueRenderer}
+        />
         <br />
         <ul className='list-group'>
-          {map(props.log.calls, call => {
+          {map(this._getFilteredCalls(), call => {
```

Comment:
Same here. Use a selector, otherwise, `_.filter` will be run on each render on (potentially) a big collection.